### DISCUSSION TITLE
libcurl: is there any way I can catch response with status code 100?

### DISCUSSION THREAD

**Discussion by teocheung** [curl/curl#14346] [Q&A] [answered]
Hi.

I'm trying to write a client demo to do http request and handle response with curl_easy.

I wanna send a post request with header "Expect: 100-continue" and "Content-Length: 17" but no content to http://postman-echo.com/post to do my test. I expect curl_easy_perform would success and return 100 status code so that I can do other actions and later send the content to server in request body. However, with CURLOPT_VERBOSE on, I do find 100-continue sending back from server, but the program is hanging in curl_easy_perform. Seems like it is still waiting for data to send, but I'm aiming to receive 100 code and do some other things and later send data.

Is there any way I can catch response with status code 100?

Here is my code:
```#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <curl/curl.h>

int main(void) {
    CURL *curl;
    CURLcode res;

    // Initialize a CURL easy session
    curl_global_init(CURL_GLOBAL_DEFAULT);
    curl = curl_easy_init();
    if(curl) {
        long long_status;
        const char *url = "http://postman-echo.com/post";

        struct curl_slist *headers = NULL;
        headers = curl_slist_append(headers, "Expect: 100-continue");
        headers = curl_slist_append(headers, "Content-Length: 17");

        // Set URL for the operation
        curl_easy_setopt(curl, CURLOPT_URL, url);

        // Set HTTP headers
        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);

        curl_easy_setopt(curl, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);

        curl_easy_setopt(curl, CURLOPT_POST, 1);

        // Verbose output for debugging
        curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
        curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, 5);
        curl_easy_setopt(curl, CURLOPT_TIMEOUT_MS, 8 * 1000L);

        // Perform the request, res will get the return code
        res = curl_easy_perform(curl);

        // Check for errors
        if(res != CURLE_OK) {
            fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
        }

        curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &long_status);
        fprintf(stderr, "status_code: %d\n", long_status);
        if (long_status == 100L) {
            // do something
        }


        // Cleanup
        curl_slist_free_all(headers);
        curl_easy_cleanup(curl);
    }

    curl_global_cleanup();
    return 0;
}
```

And here is the info from CURLOPT_VERBOSE  on:
```* About to connect() to postman-echo.com port 80 (#0)
*   Trying 3.229.206.27...
* Connected to postman-echo.com (3.229.206.27) port 80 (#0)
> POST /post HTTP/1.1
Host: postman-echo.com
Accept: */*
Expect: 100-continue
Content-Length: 17
Content-Type: application/x-www-form-urlencoded

< HTTP/1.1 100 Continue
```

Thanks!

**Comment by bagder** [comment, marked as answer]
100 is not the final response code, it is just *transitional* so curl does not stop there, it will wait for the next response code and continue its operation.

**Reply by teocheung** [reply]
Thanks Daniel. So there is no option or no way to catch 100 with libcurl right?

**Reply by bagder** [reply]
What do you mean by "catching" it? You mean figuring out that there is a 100 response?

**Reply by teocheung** [reply]
Yes. Like I could know there is a 100 response before I continue to send request body.

**Reply by bagder** [reply]
That's not how libcurl works with 100-continue. You tell libcurl what to send, and if gets a 100 it will continue and send it. If it gets an error instead (without the timeout it waits for the 100), it will not send the data.
